src/rgw/rgw_acl.cc

```diff
@@ -166,21 +166,29 @@ uint32_t RGWAccessControlPolicy::get_perm(const DoutPrefixProvider* dpp,
   ldpp_dout(dpp, 20) << "-- Getting permissions begin with perm_mask=" << perm_mask
                  << dendl;
 
-  uint32_t perm = acl.get_perm(dpp, auth_identity, perm_mask);
-
-  if (auth_identity.is_owner_of(owner.id)) {
-    perm |= perm_mask & (RGW_PERM_READ_ACP | RGW_PERM_WRITE_ACP);
-  }
+  uint32_t perm = 0;
+  const uint32_t user_perm = auth_identity.get_perm_mask();
+  if (user_perm == RGW_PERM_FULL_CONTROL) {
+    // Only consider ACL if the user has full control (either a subuser with full control or a non-subuser)
+    perm = acl.get_perm(dpp, auth_identity, perm_mask);
+
+    if (auth_identity.is_owner_of(owner.id)) {
```

Comment:
Regarding the recent account implementation, this probably needs to be replaced by something like this: `is_root_of()` https://github.com/ceph/ceph/pull/57629/commits/06a26454a57f9916a40d88be62694470bde2f952#diff-086b6cb87149dc85d4a1160169c614771a444abeefd5cb6730a3bf65155df825R64

Reply:
@cbodley Could you take a look at this?